lib/internal/modules/run_main.js

```diff
@@ -12,19 +12,33 @@ const path = require('path');
  * @param {string} main - Entry point path
  */
 function resolveMainPath(main) {
-  // Note extension resolution for the main entry point can be deprecated in a
-  // future major.
-  // Module._findPath is monkey-patchable here.
-  const { Module } = require('internal/modules/cjs/loader');
-  let mainPath = Module._findPath(path.resolve(main), null, true);
+  if (getOptionValue('--preserve-symlinks-main')) { return; }
```

Comment:
I don't think this is correct, you still want to call `Module._findPath` when the default type is `'commonjs'`. It's surprising that there are no tests to catch that.

Reply:
I know, this is why I was so surprised at the change you wanted to make. Can we just go back to the "unoptimized" version, where I'm not changing unrelated logic? And we can refactor this for performance in a separate PR.

Reply:
I disagree it's unrelated, but do whatever suits you